src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -935,11 +947,26 @@ private void ParseOption(string name, string value)
                 case "sockettimeoutms":
                     _socketTimeout = ParseTimeSpan(name, value);
                     break;
-                case "ssl":
-                    _ssl = ParseBoolean(name, value);
+                case "ssl": // Obsolete
+                case "tls":
+                    var tlsValue = ParseBoolean(name, value);
+                    if (_tls.HasValue && _tls.Value != tlsValue)
+                    {
+                        throw new MongoConfigurationException("tls has already been configured with a different value.");
+                    }
+                    _tls = tlsValue;
                     break;
-                case "sslverifycertificate":
-                    _sslVerifyCertificate = ParseBoolean(name, value);
+                case "sslverifycertificate": // Obsolete
+                    {
+                        var tlsInsecureValue = !ParseBoolean(name, value);
+                        _tlsInsecure = ParseTlsInsecure(tlsInsecureValue);
+                    }
+                    break;
```

Comment:
While there is nothing inherently wrong with brackets nested inside a case, they are often not necessary. In this case a simple variable renaming would make them unnecessary:

    case "sslverifycertificate": // Obsolete
        var sslVerifyCertificateValue = ParseBoolean(name, value);
        _tlsInsecure = EnsureTlsInsecureIsValid(!sslVerifyCertificateValue);
        break;